VultisigApp/VultisigApp/Services/RpcEvmService.swift

```diff
@@ -16,17 +16,42 @@ class RpcEvmService: RpcService {
     
     func getBalance(coin: Coin) async throws ->(rawBalance: String,priceRate: Double){
         // Start fetching all information concurrently
-        let cryptoPrice = await CryptoPriceService.shared.getPrice(priceProviderId: coin.priceProviderId)
+        var cryptoPrice = Double(0)
         var rawBalance = ""
         do{
+            if !coin.priceProviderId.isEmpty {
+                cryptoPrice = await CryptoPriceService.shared.getPrice(priceProviderId: coin.priceProviderId)
+            }
+            
             if coin.isNativeToken {
                 rawBalance = String(try await fetchBalance(address: coin.address))
             } else {
                 rawBalance = String(try await fetchERC20TokenBalance(contractAddress: coin.contractAddress, walletAddress: coin.address))
+                
+                // Probably a custom token, let's try to get the price from the pool
```

Comment:
I think RPCEvmService should not care about type of Coin (custom token or native coin from our token store)

Reply:
What is your suggestion?